### PR TITLE
chore(prod): refactor the production build

### PR DESCRIPTION
**Pull request by mgechev**
Main changes:

- Moved the production build from `System.register` format with the `systemjs-builder` to browserify. The main purpose is to reduce the bundle size.
- Removes systemjs' config
- Creates a single bundle for all libraries and the application itself (except the shims which at this point are into a separate file, included before the app)
- Minifies all the CSS resources and bundles them into a single file
- Copies all the files different from `.ts` for `dev` and `.ts`, `.html`, `.css` for `prod` (since the templates are supposed to be inlined, the CSS files are either or bundled)
- Adds another array of shims required for the production build

Fixes https://github.com/mgechev/angular2-seed/issues/383, https://github.com/mgechev/angular2-seed/issues/364, https://github.com/mgechev/angular2-seed/issues/362, https://github.com/mgechev/angular2-seed/issues/350.

### CODE REVIEW

File: tools/utils/template_locals.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
-import {APP_BASE, APP_DEST, APP_ROOT, APP_TITLE, SYSTEM_CONFIG, VERSION, HOT_LOADER_PORT, BOOTSTRAP_MODULE} from '../config';
+import {APP_BASE, APP_DEST, APP_ROOT, APP_TITLE, SYSTEM_CONFIG, VERSION, HOT_LOADER_PORT, BOOTSTRAP_MODULE, ENV} from '../config';
```
Comment (ludohenin) [inline tools/utils/template_locals.ts]: Don't you think we should `import * as CONFIG from ...` and return `CONFIG` so we never have to touch this ?

File: tools/config.ts [hunk 2]
```diff
@@ -80,14 +93,7 @@ const SYSTEM_CONFIG_DEV = {
...
-  bundles: {
-    'bundles/app': ['bootstrap']
-  }
-};
-
-export const SYSTEM_CONFIG = ENV === 'dev' ? SYSTEM_CONFIG_DEV : SYSTEM_CONFIG_PROD;
+export const SYSTEM_CONFIG = SYSTEM_CONFIG_DEV;
 
 // This is important to keep clean module names as 'module name == module uri'.
 export const SYSTEM_CONFIG_BUILDER = {
```
Comment (ludohenin) [inline tools/config.ts]: I guess this is not required anymore.

File: tools/tasks/build.assets.dev.ts [hunk 3]
```diff
@@ -4,12 +4,8 @@ import {APP_SRC, APP_DEST} from '../config';
 export = function buildImagesDev(gulp, plugins) {
```
Comment (ludohenin) [inline tools/tasks/build.assets.dev.ts]: function names isn't correct :)

File: tools/tasks/build.assets.prod.ts [hunk 4]
```diff
@@ -2,13 +2,25 @@ import {join} from 'path';
 import {APP_SRC, APP_DEST} from '../config';
 
 export = function buildImagesDev(gulp, plugins) {
```
Comment (ludohenin) [inline tools/tasks/build.assets.prod.ts]: function names isn't correct :)

File: tools/config.ts [hunk 5]
```diff
@@ -29,44 +34,52 @@ export const TEST_DEST            = 'test';
...
+  { src: 'systemjs/dist/system.src.js', inject: 'shims', dest: JS_DEST },
+  { src: 'angular2/bundles/angular2-polyfills.js', inject: 'shims', dest: JS_DEST },
+  { src: 'rxjs/bundles/Rx.js', inject: 'libs', dest: JS_DEST },
+  { src: 'angular2/bundles/angular2.js', inject: 'libs', dest: JS_DEST },
+  { src: 'angular2/bundles/router.js', inject: 'libs', dest: JS_DEST },
+  { src: 'angular2/bundles/http.js', inject: 'libs', dest: JS_DEST },
+  { src: 'bootstrap/dist/css/bootstrap.css', inject: true, dest: CSS_DEST }
+]);
+
+export const PROD_NPM_DEPENDENCIES: InjectableDependency[] = normalizeDependencies([
```
Comment (ludohenin) [inline tools/config.ts]: `angular2-polyfills.js` is missing here (zone not included, so change detection doesn't work in about page)

File: tools/utils/template_locals.ts (hunk 1 above)
Comment (mgechev) [inline tools/utils/template_locals.ts]: Yep, makes sense.

File: tools/config.ts [hunk 6]
```diff
@@ -29,44 +34,53 @@ export const TEST_DEST            = 'test';
...
+  { src: 'reflect-metadata/Reflect.js', inject: 'shims' },
+  { src: 'es6-shim/es6-shim.min.js', inject: 'shims' },
+  { src: 'angular2/bundles/angular2-polyfills.min.js', inject: 'libs' },
```
Comment (mgechev) [inline tools/config.ts]: @ludohenin thanks for the catch! I added it with the latest update.

**Pull request merged**